script/Amex-Select-Offers.js

```diff
@@ -90,22 +98,26 @@
           setTimeout(() => {
             addToCardButton.click();
             console.log(`Clicked "${offerName}" offer!`);
-          }, i * WAIT_TIME_MS);
+          }, WAIT_TIME_MS);
         } else {
           console.log(`Button for offer ${i} is not Add to Card, skipping...`);
         }
       } catch (e) {
         console.error(e);
       }
-    }
 
-    console.log("Done selecting offers!");
-    disableButton();
+      if (this.doneFlag) {
+        disableButton();
+        console.log("Done selecting offers!");
+      } else {
+        inProgressButton();
+      }
+    }

```

Comment:
personally no need to have additional checks inside a loop, you can keep them outside. The `doneFlag` can be a local variable as well if preferred.

Reply:
This is what I had done for the time being. Not exactly ready which is why I hadn't said it was truly good to go.